msteams-platform/bots/how-to/Teams conversational AI/teams-conversation-ai-overview.md

```diff
@@ -75,29 +100,57 @@ A simple interface for actions and predictions allows bots to react when the bot
 
-* Support to OAuth S2S
-* Adherence to Activity schema for reading and writing JSON documents
-* Invoking Rest APIs to determine additional context required to handle a user's message, such as Azure Active Directory (Azure AD) ID and UPN of the user the bot is interacting with. -->
+* **Monologue**: Create AutoGPT-style agents capable of performing multi-step actions independently and reliability with full schema validation and automatic repair included.
+
+* **Sequences**: Enable your AI assistant to return a sequence of actions for execution with schema validation increasing reliability.
+
+* **Functions**: Produce structured responses from your model by employing user-defined Functions. These functions are customizable using JSON schemas to define the parameters and their format. The ActionPlanner assesses model responses against the schema, making repairs as needed increasing response reliability and consistency.
+
+### Vector data sources
+
+Vector Databases are a novel type of database designed specifically for storing Vectors and facilitating efficient search over them. They return the most relevant results for a user’s query. The vector search feature in a vector database enables retrieval-augmented generation to harness LLMs and custom data or domain-specific information. This process involves extracting pertinent information from a custom data source and integrating it into the model request through prompt engineering. Prior to sending a request to the LLM, the user input/query/request is also transformed into an embedding, and vector search techniques are employed to locate the most similar embeddings within the database.
+
+## Enhanced reasoning
+
+With enhanced reasoning, the Teams AI Library tackles the ongoing issue of bot hallucinations head-on by offering an integrated fact-checking system.
+
+When a user interacts with your AI assistant, this system prompts the bot to engage in a process of self-reflection critically evaluating its potential responses prior to sending them. The introspection allows the bot to identify inaccuracies and correct its answers improving accuracy, quality, and contextual relevance. Advanced reasoning ensures that your AI assistant becomes a dependable source of information and judgment, building trust in your product and drawing users back every day.
+
+## Feedback loop
+
+The feedback loop is a feature of the AI system that allows it to validate and correct the output of the language model. It works by checking the structure and parameters of the plan or monologue that the model returns, and giving it feedback if there are any errors or missing information. The model then tries to fix its mistakes and return a valid output. The feedback loop can improve the reliability and accuracy of the AI system, and reduce the chances of hallucination or invalid actions.
+
+The following table lists the updates to the Teams AI library:
+
+|Type |Description  |JavaScript|C#|
```

Comment:
the table is not generated properly for me in the github markdown preview

Reply:
The table will not be generated in the markdown preview, we need to check the review site for table update: https://review.learn.microsoft.com/en-us/microsoftteams/platform/bots/how-to/teams%20conversational%20ai/teams-conversation-ai-overview?branch=pr-en-us-10013